### ISSUE TITLE
Preview avatar point of view while using avatar builder

### ISSUE THREAD

**Issue by Gawdl3y** [Neos-Metaverse/NeosPublic#1818]
## Is your feature request related to a problem? Please describe.
When creating an avatar, it can be difficult to get the position of the HMD in a good place for the avatar's point of view to be just right. Often, the avatar builder has to be redone or the point of view adjusted in the inspector after the fact.

## Relevant issues
I wasn't able to find any with the keywords I was using.

## Describe the solution you'd like
A button to temporarily move your current viewpoint to the avatar builder's HMD position would be great, and would allow for quickly previewing what you would see from it before finishing the avatar builder process. Perhaps in addition to that, when the avatar builder is open, a 2D camera preview of the avatar's perspective popping up would help. With that, a selector for which eye it renders as (left, right, both side-by-side) would likely be necessary.

## Describe alternatives you've considered
The ability to open the avatar builder on an already-built avatar to make changes would also make tweaking this much easier/more obvious.

## Additional context
Original idea by Ari_ / Ari_#5423. He brought this up in conversation with me, and I agreed it would be very useful, thus created this issue.

**Comment by ohzee00** [comment]
This seems like a really good idea! Though I would highly suggest making sure there is a user friendly way to get out of said view, cause changing the viewport of a user in VR suddenly is very _very_ strange and discombobulating.

**Comment by AshtonSparx** [comment]
One way would be a low res camera built into the yellow headset you place on the model. Since different headsets have different FOV - on the builder, there could be selections on which headset you'll be using that changes the FOV to match yours. When on said selections (Shouldn't be on by default for optimization purposes) It's preview will be displayed on the builder's GUI.

**Comment by InsaneGrox** [comment]
Unfortunately, different headsets also have different view origins (fors some reason), this is why a lot of my avatars in my public folder have clipping issues on oculus headsets (and even vive headsets to a certain extent), as I only test them with an index, which has the furthest forward viewpoint.

**Comment by AshtonSparx** [comment]
I'm sure there would be a way they could also offset that with the selected headset so the preview can accurately display the desired preview

**Comment by shiftyscales** [comment]
> 
> 
> > One way would be a low res camera built into the yellow headset you place on the model. Since different headsets have different FOV - on the builder, there could be selections on which headset you'll be using that changes the FOV to match yours. When on said selections (Shouldn't be on by default for optimization purposes) It's preview will be displayed on the builder's GUI.
> 
> Unfortunately, different headsets also have different view origins (for some reason), this is why a lot of my avatars in my public folder have clipping issues on oculus headsets (and even vive headsets to a certain extent), as I only test them with an index, which has the furthest forward viewpoint.

I believe you are referring to #828 which was resolved by a patch for SteamVR. All headsets (based on my own experience in using them) should now have very similar if not identical view points regardless of runtime. Individual headsets will have varying field of views though, so a headset with a wider FOV might see clipping, etc. that the original creator couldn't when producing the item as an example.

**Comment by AshtonSparx** [comment]
> I believe you are referring to #828 which was resolved by a patch for SteamVR. All headsets (based on my own experience in using them) should now have very similar if not identical view points regardless of runtime. Individual headsets will have varying field of views though, so a headset with a wider FOV might see clipping, etc. that the original creator couldn't when producing the item as an example.

Which like I mentioned, could be adjusted to get the most accurate preview - per headset that's selected :)

**Comment by Frooxius** [comment]
Yeah this could be a good thing to add, but I'd bundle it with some bigger Avatar Creator overhaul/rework that I want to do, it's going to shake up a bunch of things and make it a fair bit easier to extend. Not sure when exactly it'll come yet.
